LongReadRNABenchmark/LongReadRNABenchmarkTasks.wdl

```diff
@@ -0,0 +1,697 @@
+        File referenceGenome
+        File referenceGenomeIndex
+        File referenceAnnotation
+        String datasetName
+        Int numThreads
+    }
+
+    String docker = "brookslab/flair:latest"
+    Int cpu = 16
+    Int memory = 256
+    Int diskSizeGB = 500
+    File monitoringScript = "gs://broad-dsde-methods-tbrookin/cromwell_monitoring_script2.sh"
+
+    String flairPrefix = "Flair_out_~{datasetName}"
+
+    command <<<
+        bash ~{monitoringScript} > monitoring.log &
+
+        samtools fastq ~{inputBAM} > "~{flairPrefix}_temp.fastq"
+
+        bam2Bed12 -i ~{inputBAM} > "~{flairPrefix}.bed"
+
+        flair correct \
+        -q "~{flairPrefix}.bed" \
+        -g ~{referenceGenome} \
+        -f ~{referenceAnnotation} \
+        -o ~{flairPrefix} \
+        -t ~{numThreads}
+
+        flair collapse \
+        -g ~{referenceGenome} \
+        -f ~{referenceAnnotation} \
+        -r "~{flairPrefix}_temp.fastq" \
+        -q "~{flairPrefix}_all_corrected.bed" \
+        -o ~{flairPrefix} \
+        -t ~{numThreads}
+
+        mv Flair_out_~{datasetName}.isoforms.gtf Flair_out_~{datasetName}.gtf
+    >>>
+
+    output {
+        File flairGTF = "Flair_out_~{datasetName}.gtf"
+        File monitoringLog = "monitoring.log"
+    }
+
+    runtime {
+        docker: docker
+        cpu: cpu
+        memory: "~{memory} GiB"
+        disks: "local-disk ~{diskSizeGB} HDD"
+    }
+}
+
+task Talon {
+    input {
+        File inputBAM
+        File inputBAMIndex
+        File referenceGenome
+        File referenceGenomeIndex
+        File referenceAnnotation
+        String datasetName
+        String dataType
+        Int numThreads
+    }
+
+    String docker = "us.gcr.io/broad-dsde-methods/kockan/talon:latest"
+    Int cpu = 16
+    Int memory = 256
+    Int diskSizeGB = 500
+    File monitoringScript = "gs://broad-dsde-methods-tbrookin/cromwell_monitoring_script2.sh"
+
+    String talonPrefix = "Talon_out_~{datasetName}"
+
+    command <<<
+        bash ~{monitoringScript} > monitoring.log &
+
+        talon_label_reads --f ~{inputBAM} --t 1 --o ~{talonPrefix} --g ~{referenceGenome}
+        samtools calmd -@ ~{numThreads} --reference ~{referenceGenome} "~{talonPrefix}_labeled.sam" > "~{talonPrefix}_labeled.md.sam"
+        talon_initialize_database --f ~{referenceAnnotation} --g ~{datasetName} --a ~{datasetName} --o ~{datasetName}
+        echo ~{datasetName},~{datasetName},~{dataType},"~{talonPrefix}_labeled.md.sam" > "~{talonPrefix}.csv"
+        talon --build ~{datasetName} --db "~{datasetName}.db" --o "~{talonPrefix}_raw" --f "~{talonPrefix}.csv" --threads ~{numThreads}
+        talon_filter_transcripts --db "~{datasetName}.db" -a ~{datasetName} --datasets ~{datasetName} --o "~{talonPrefix}_filter"
+        talon_create_GTF --build ~{datasetName} --db "~{datasetName}.db" -a ~{datasetName} --o ~{talonPrefix} --whitelist "~{talonPrefix}_filter"
+
+        mv Talon_out_~{datasetName}_talon.gtf Talon_out_~{datasetName}.gtf
```

Comment:
this doesn't do anything, does it?

Reply:
There is an "_talon" at the end of the first file. Similar to some other outputs, this renaming is just for consistency and we might choose not to do it or do it elsewhere. Looks nicer in data table outputs but not required.